sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -4255,11 +4255,11 @@ object functions {
    *
    * @group udf_funcs
    * @since 2.0.0
+   *
+   * @deprecated("please use the typed `udf` methods", "3.0.0")
```

Comment:
with Scala 2.12, type and nullability info need to be retrieved during compile time, this method is not very useful and we should deprecate it.